.github/workflows/builder.yaml

```diff
@@ -36,15 +39,16 @@ jobs:
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: 🛠️ Build devcontainer (${{ matrix.devcontainer }})
+      - name: 🛠️ Build devcontainer (${{ matrix.devcontainer }}, ${{ matrix.platform }})
         uses: docker/build-push-action@v2
         with:
+          load: true
           context: .
           file: ./${{ matrix.devcontainer }}/Dockerfile
-          platforms: linux/amd64,linux/arm64
+          platforms: ${{ matrix.platform }}
```

Comment:
This will push each platform to the same tag, meaning the latest platform that builds will overwrite the first.
`load: true` is the only thing needed in this PR as for as I can tell

Reply:
@ludeeus That's what I tried in the first commit, the result was [this](https://github.com/home-assistant/devcontainer/runs/5330002050?check_suite_focus=true):
```
/usr/bin/docker buildx build --file ./addons/Dockerfile --iidfile /tmp/docker-build-push-jvKiY3/iidfile --platform linux/amd64,linux/arm64 --tag ghcr.io/home-assistant/devcontainer:addons --load --metadata-file /tmp/docker-build-push-jvKiY3/metadata-file .
error: docker exporter does not currently support exporting manifest lists
Error: buildx failed with: error: docker exporter does not currently support exporting manifest lists
```
It looks like `--load` doesn't work with multi-architecture images, you can only push them. What if we just did this instead?
```yaml
 - name: 🛠️ Build devcontainer (${{ matrix.devcontainer }}, ${{ matrix.platform }})
    uses: docker/build-push-action@v2
    with:
      push: ${{ github.event_name == 'push' }}
      context: .
      file: ./${{ matrix.devcontainer }}/Dockerfile
      platforms: ${{ matrix.platform }}
      tags: ghcr.io/${{ github.repository_owner }}/devcontainer:${{ matrix.devcontainer }}
```
And then just dropped the publish step entirely since this one does both?

Reply:
Oh sorry, my bad I meant this:
```yaml
- name: 🛠️ Build devcontainer (${{ matrix.devcontainer }}, ${{ matrix.platform }})
    uses: docker/build-push-action@v2
    with:
      push: ${{ github.event_name == 'push' }}
      context: .
      file: ./${{ matrix.devcontainer }}/Dockerfile
      platforms: linux/amd64,linux/arm64
      tags: ghcr.io/${{ github.repository_owner }}/devcontainer:${{ matrix.devcontainer }}
```
Drop the second matrix, back to what we had just build and push all in one action when the event is `push`

Reply:
That should work 👍